types/big.js/index.d.ts

```diff
@@ -1,412 +1,359 @@
+         * @param rm Rounding mode: 0 (down), 1 (half-up), 2 (half-even) or 3 (up).
+         * @throws `!toPre!` if sd is invalid.
+         */
+        toPrecision(sd?: number, rm?: RoundingMode): string;
+        /**
+         * Returns a string representing the value of this Big number.
+         *
+         * If this Big number has a positive exponent that is equal to or greater than 21, or a negative exponent equal to or less than -7, then exponential notation is returned.
+         *
+         * The point at which toString returns exponential rather than normal notation can be adjusted by changing
+         * the value of Big.E_POS and Big.E_NEG. By default, Big numbers correspond to Javascript's number type in this regard.
+         */
+        toString(): string;
+        /**
+         * Returns a primitive number representing the value of this Big number.
+         *
+         * If Big.strict is true an error will be thrown if toNumber is called on a Big number which cannot be converted to a primitive number without a loss of precision.
+         *
+         * @since 6.0
+         */
+        toNumber(): number;
+        /**
+         * Returns a string representing the value of this Big number.
+         *
+         * If this Big number has a positive exponent that is equal to or greater than 21, or a negative exponent equal to or less than -7, then exponential notation is returned.
+         *
+         * The point at which toString returns exponential rather than normal notation can be adjusted by changing
+         * the value of Big.E_POS and Big.E_NEG. By default, Big numbers correspond to Javascript's number type in this regard.
+         */
+        valueOf(): string;
+        /**
+         * Returns a string representing the value of this Big number.
+         *
+         * If this Big number has a positive exponent that is equal to or greater than 21, or a negative exponent equal to or less than -7, then exponential notation is returned.
+         *
+         * The point at which toString returns exponential rather than normal notation can be adjusted by changing
+         * the value of Big.E_POS and Big.E_NEG. By default, Big numbers correspond to Javascript's number type in this regard.
+         */
+        toJSON(): string;
+        /**
+         * Returns an array of single digits
+         */
+        c: number[];
+        /**
+         * Returns the exponent, Integer, -1e+6 to 1e+6 inclusive
+         */
+        e: number;
+        /**
+         * Returns the sign, -1 or 1
+         */
+        s: number;
+    }
 }
 
 // We want the exported symbol 'Big' to represent two things:
 // - The Big interface, when used in a type context.
 // - The BigConstructor instance, when used in a value context.
-export const Big: BigConstructor;
-
-// The default export is the same as type/value combo symbol 'Big'.
-export default Big;
+declare const Big: Big.BigConstructor;
+type Big = Big.Big;
```

Comment:
Isn't the normal way to do this to just write a merged declaration? i.e.:

```ts
declare function Big(): typeof Big;
declare function Big(value: Big.BigSource): Big;
declare class Big {
  constructor(value: Big.Bigsource);
  // either declare statics for DP/RM/etc. or make them namespace exports?
}
interface Big {
 ... // maybe just merge into class above? 
}
declare namespace Big {
  // everything else in Big namespace
}
export = Big;
```

We will merge the `function` and `class` declarations in the .d.ts file, so there's no need to declare a `BigConstructor` (though you could do `export type BigConstructor = typeof Big` if you wanted to)

Reply:
Yeah, I’m not sure why this was initially declared as a separate constructor. I was trying not to change stuff that I didn’t have to change as part of this refactor, but maybe that’s worthwhile.

Reply:
Oh, I think I see a reason. If you call (not construct) the “constructor” function with no arguments, it returns _another constructor function_ https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66163/files#diff-8034f96ce0019a4b4f9f3431da74d25feaa41073c0f68ca5f02f4356eb3076ffR54

I don’t think this is something a class declaration can support.

Reply:
That's why the example I posted above has both `declare function Big()` and `declare class Big`. We allow function/class merging when both declarations are ambient types.

Reply:
Oof, TIL.

Is there any reason to make that refactor now? It’s been like this for a long time.

Reply:
That's fine. I just figured if we're making a change we might as well use a cleaner style, but its not an issue.